v2/concordium/types.proto

```diff
@@ -3540,3 +3540,53 @@ message DryRunSuccessResponse {
     }
 }
 
+// A request from a light client to catch up from a trusted block to a subsequent finalized
+// block.
+message LightBlocksRequest {
+    // The highest finalized block currently trusted by the light client.
+    BlockHashInput from = 1;
+    // The target finalized block to catch the light client up to.
+    // The request will fail if this is not descended from the 'from' block.
+    BlockHashInput to = 2;
+    // If true, indicates that the light client is not aware of the current epoch finalization
+    // committee for it's highest finalized block.
+    bool current_committee_unknown = 3;
+    // If true, indicates that the light client is not aware of the next epoch finalization
+    // committee for it's highest finalized block.
+    bool next_committee_unknown = 4;
+    // If true, the response should include a proof of the next epoch finalization committee for
+    // the 'to' block.
+    bool next_committee_required = 5;
+}
+
+// Representation of a generic Merkle proof.
+// A Merkle proof can be canonically converted to a root hash by converting each of its
+// branches to bytes, concatenating them, and then taking the SHA256 hash of the result.
+message MerkleProof {
```

Comment:
**Suggestion:** On top of this, we should probably have new type wrappers for each type of merkle proof that we will expose.